hips/tiles/tile.py

```diff
@@ -0,0 +1,107 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+import urllib.request
+from io import BytesIO
+
+import numpy as np
+from PIL import Image
+from astropy.io import fits
+from astropy.io.fits.header import Header
+
+from .tile_meta import HipsTileMeta
+
+__all__ = [
+    'HipsTile',
+]
+
+__doctest_skip__ = ['HipsTile']
+
+
+class HipsTile:
+    """HiPS tile container.
+
+    This class provides methods for fetching, reading, and writing a HiPS tile.
+
+    Parameters
+    ----------
+    meta : `HipsTileMeta`
+        Metadata of HiPS tile
+    data : `~numpy.ndarray`
+        Data containing HiPS tile
+    header : `~astropy.io.fits.Header`
+        Header of HiPS tile
+
+    Examples
+    --------
+        >>> from hips.tiles import HipsTile
+        >>> from hips.tiles import HipsTileMeta
+        >>> meta = HipsTileMeta(order=6, ipix=30889, format='fits')
+        >>> hips_tile = HipsTile.read(meta)
+        >>> hips_tile.data
+        array([[0, 0, 0, ..., 0, 0, 0],
+               [0, 0, 0, ..., 0, 0, 0],
+               [0, 0, 0, ..., 0, 0, 0],
+               ...,
+               [0, 0, 0, ..., 1, 0, 0],
+               [0, 0, 0, ..., 1, 0, 1],
+               [0, 0, 0, ..., 1, 0, 1]], dtype=int16)
+        """
+
+    def __init__(self, meta: HipsTileMeta, data: np.ndarray = None, header: Header = None) -> None:
+        self.meta = meta
+        self.format = format
```

Comment:
This is a bug, you have to remove the line `self.format = format`.
The reason it isn't faliing in your tests is that `format` is a Python built-in function, i.e. exists.
That's the reason one shouldn't use variable names that match Python builtins, something a linter like pyflakes or pylint warns about.
Sometimes there's a tradeoff and you have to decide if to pick the nice name `format` or change e.g. to `fmt` to avoid this potential issue (a choice you have to make in `HipsDataMeta`). My suggestion would be to change to `file_format` in this case.